### PR TITLE
feat(frontend): Show Notification

### LINKED ISSUE #468: Show notifications
- [x] @murillojorge can you help us with a template for notifications?
- [x] Define design for notifications
- [x] Add categories #544 
- [x] Add microcopies #548 
- [x] Add design #549 @steph222 
- [x] Apply changes to database @leisterfrancisco 

Dependencies:
#544 
#548 
#549 

image

Users can receive push-notifications on their phones when they donate blood 

Own: @steph222 
Sup: @leisterfrancisco and @Edgar097

### PR DESCRIPTION
**Pull request by steph222**
Resolves #468 
Remember to stop and delete docker containers, before testing

### CODE REVIEW

File: webapp/src/components/Notification/Notification.js [hunk 1]
```diff
@@ -15,6 +15,17 @@ import useMediaQuery from '@material-ui/core/useMediaQuery'
 import { useTranslation } from 'react-i18next'
 import { useLocation } from 'react-router-dom'
 
+
```
Comment (tetogomez) [inline webapp/src/components/Notification/Notification.js:18]: Remember, you just need one line to separate import groups

File: webapp/src/components/Notification/Notification.js [hunk 2]
```diff
@@ -15,6 +15,17 @@ import useMediaQuery from '@material-ui/core/useMediaQuery'
...
+import { useSubscription } from '@apollo/react-hooks'
+import { useUser } from '../../context/user.context'
+import NotificationStructure from '../NotificationStructure'
+
+import {
+  NOTIFICATION_SUBSCRIPTION
+} from '../../gql'
+
+
```
Comment (tetogomez) [inline webapp/src/components/Notification/Notification.js:27]: remove unnecessary blank spaces

File: webapp/src/components/Notification/Notification.js [hunk 3]
```diff
@@ -29,6 +40,7 @@ const useStyles = makeStyles((theme) => ({
   },
   appBar: {
     position: 'relative',
+
```
Comment (tetogomez) [inline webapp/src/components/Notification/Notification.js:43]: same here

File: webapp/src/components/Notification/Notification.js [hunk 4]
```diff
@@ -75,7 +98,27 @@ const Notification = () => {
     setOpen(false)
   }
 
+  const { data: notification = {} } = useSubscription(
+    NOTIFICATION_SUBSCRIPTION, { variables: { account } }
+  )
+
```
Comment (tetogomez) [inline webapp/src/components/Notification/Notification.js:104]: and here

File: webapp/src/components/Notification/Notification.js [hunk 5]
```diff
@@ -75,7 +98,27 @@ const Notification = () => {
...
+
+  useEffect(() => {
+
```
Comment (tetogomez) [inline webapp/src/components/Notification/Notification.js:107]: same here, unnecessary blank spaces

File: webapp/src/components/Notification/Notification.js [hunk 6]
```diff
@@ -75,7 +98,27 @@ const Notification = () => {
...
+
+    if (Object.keys(notification).length) {
+      const notificationList = notification.notification
+      console.log("Entro...")
```
Comment (tetogomez) [inline webapp/src/components/Notification/Notification.js:111]: this console logs are necessary ??

File: webapp/src/components/Notification/Notification.js [hunk 7]
```diff
@@ -85,16 +128,18 @@ const Notification = () => {
           })}
         />
       </IconButton>
-      <Dialog
-        fullScreen
+
+      <Box className={classes.box}
+
```
Comment (tetogomez) [inline webapp/src/components/Notification/Notification.js:133]: please add correct indentation for Box props

File: webapp/src/components/Notification/Notification.js [hunk 8]
```diff
@@ -105,8 +150,21 @@ const Notification = () => {
             </Typography>
           </Toolbar>
         </AppBar>
-        <List />
-      </Dialog>
+        {notifications.length > 0 && (
+          <>
+            {notifications.map((el, key) => (
+
```
Comment (tetogomez) [inline webapp/src/components/Notification/Notification.js:156]: unnecessary blank space

File: webapp/src/components/NotificationStructure/NotificationStructure.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+      <Box >
+        <Typography>
+          {title}
+        </Typography>
+      </Box>
+
+    </>
+  )
+}
+
```
Comment (tetogomez) [inline webapp/src/components/NotificationStructure/NotificationStructure.js:20]: same here

File: webapp/src/components/NotificationStructure/NotificationStructure.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+NotificationStructure.propTypes = {
+  title: PropTypes.string,
+  description: PropTypes.string
+
```
Comment (tetogomez) [inline webapp/src/components/NotificationStructure/NotificationStructure.js:25]: and here

**Review by tetogomez — CHANGES_REQUESTED** [review]
Please check comments

File: webapp/src/components/Notification/Notification.js (hunk 1 above)
Comment (steph222) [reply webapp/src/components/Notification/Notification.js:18]: Disculpa @tetogomez , aún no era la versión final, es que Leister me lo había solicitado para utilizar una parte del código, por lo implemente la etiquete WIP, gracias por las observaciones!!

File: hapi/src/routes/edit-notification-state/edit-notification-state.handler.js [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+const { BAD_REQUEST } = require('http-status-codes')
+
+const { donorApi, sponsorApi, lifebankApi } = require('../../api')
+
+module.exports = async ({ auth: { credentials }, payload: { input } }) => {
+  try {
+    const role =
+      credentials['https://hasura.io/jwt/claims']['x-hasura-default-role']
+
+    console.log('ID', input)
```
Comment (leisterfrancisco) [inline hapi/src/routes/edit-notification-state/edit-notification-state.handler.js:11]: Remove console log

File: hapi/src/routes/edit-notification-state/edit-notification-state.handler.js [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+    console.log('CREDENTIALS', credentials.sub)
```
Comment (leisterfrancisco) [inline hapi/src/routes/edit-notification-state/edit-notification-state.handler.js:12]: Remove console log

File: hapi/src/routes/edit-notification-state/edit-notification-state.handler.js [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+    //   /* case 'sponsor':
+    //      await sponsorApi.editProfile(credentials.sub, input.profile)
+    //      break
+    //    case 'lifebank':
+    //      await lifebankApi.editProfile(credentials.sub, input.profile)
+    //      break
+    //    default:
+    //      throw new Error('Invalid role')
+    //      break*/
+    // }
```
Comment (leisterfrancisco) [inline hapi/src/routes/edit-notification-state/edit-notification-state.handler.js:27]: Take a look at comments

File: webapp/src/components/NotificationStructure/NotificationStructure.js [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+    editNotificationState({
+      variables: {
+        id: id
+      }
+    })
+  }
+
+  return (
+    <>
+
```
Comment (leisterfrancisco) [inline webapp/src/components/NotificationStructure/NotificationStructure.js:51]: Remove empty line

**Review by leisterfrancisco — CHANGES_REQUESTED** [review]

**Review by Edgar097 — APPROVED** [review]

**Comment by leisterfrancisco** [comment]
@steph222 don't forget to add a description to understand rapidly what does this RP do?

**Review by leisterfrancisco — APPROVED** [review]

**Pull request merged**